src/rules/markUsed.js

```diff
@@ -0,0 +1,98 @@
+import {
+  parse,
+} from 'jsdoc-type-pratt-parser';
+import iterateJsdoc from '../iterateJsdoc';
+
+/**
+ * Extracts the type names from parsed type declaration.
+ */
+const extractTypeNames = (parsed) => {
+  if (typeof parsed !== 'object') {
+    /* istanbul ignore next */
+    return [];
+  } else if (parsed.type === 'JsdocTypeName') {
```

Comment:
Could you change everything starting with this line to a `switch`? I know it's less capable if needing expanding, but it helps convey more quickly that this is mostly comparing against a single variable.

Reply:
Yeah, go ahead for now.

Reply:
Pushed 👍